### PR TITLE
Third step of Stone duality for spectral locales

### LINKED ISSUE #269: Add alternative definition of the notion of distributive lattice isomorphism
As pointed by @martinescardo at the meeting on 2024-04-30.

### PR DESCRIPTION
**Pull request by ayberkt**
This PR implements the third step towards Stone duality for spectral locales: the type equivalence between large, locally small, and small cocomplete spectral locales and small distributive lattices.

This PR also fixes #269 in passing.

The fourth and the final step will be the extension of this to a categorical equivalence.

### CODE REVIEW

File: source/Locales/Adjunctions/Properties.lagda [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   open PosetReasoning Q
+
+   † : (s x ≤[ Q ] y ⇒ x ≤[ P ] r y) holds
+   † p = sections-are-order-embeddings P Q s r 𝓂₂ ψ ※
+    where
+     Ⅰ = p
+     Ⅱ = φ y ⁻¹
+
+     ※ : (s x ≤[ Q ] s (r y)) holds
+     ※ = s x ≤⟨ Ⅰ ⟩ y ＝⟨ Ⅱ ⟩ₚ s (r y) ■
```
Comment (martinescardo) [inline source/Locales/Adjunctions/Properties.lagda]: Make this to be several lines.

File: source/Locales/Adjunctions/Properties.lagda [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+   ‡ : (x ≤[ P ] r y ⇒ s x ≤[ Q ] y) holds
+   ‡ p = s x ≤⟨ Ⅰ ⟩ s (r y) ＝⟨ Ⅱ ⟩ₚ y ■
```
Comment (martinescardo) [inline source/Locales/Adjunctions/Properties.lagda]: And this too.

File: source/Locales/DistributiveLattice/Ideal.lagda [hunk 3]
```diff
@@ -184,6 +187,8 @@ module PrincipalIdeals (L : DistributiveLattice 𝓤) where
 
 \end{code}
 
+Some nice syntax.
+
 \begin{code}
 
  syntax principal-ideal x = ↓ x
```
Comment (martinescardo) [inline source/Locales/DistributiveLattice/Ideal.lagda]: I don't think `syntax` is needed for this.

File: source/Locales/StoneDuality/ForSpectralLocales.lagda [hunk 4]
```diff
@@ -143,3 +148,63 @@ spectral-with-small-basis-iff-spectral· X = † , ‡
...
+Added on 2024-06-20.
+
+We now show that the type `Spectral-Locale 𝓤` is equivalent to the type
+`DistributiveLattice 𝓤`.
+
+Recall that the type of spectral locales is defined as:
+
+\begin{code}
+
+Spectral-Locale : (𝓤 : Universe) → 𝓤 ⁺ ⁺  ̇
```
Comment (martinescardo) [inline source/Locales/StoneDuality/ForSpectralLocales.lagda:161]: Remove the space between the pluses.

**Review by martinescardo — APPROVED** [review]
There are minor things to maybe change.

File: source/Locales/DistributiveLattice/Ideal.lagda (hunk 3 above)
Comment (ayberkt) [reply source/Locales/DistributiveLattice/Ideal.lagda:196]: Tried turning this into a definition with the same precedence, but it did not work. I will fix it in the future.

**Pull request merged**